### PR TITLE
add DEBUG flag to EthOptions.cmake && remove useless functions and micro-definitions of Common.cpp of libdevcore && add unit tests for Common.* of libdevcore

### PR DESCRIPTION
**Pull request by cyjseagull**
1.add DEBUG to EthOptions.cmake && add ETH_DEBGU and ETH_TESTS definition when either DEBUG and TESTS flag is setted
2. remove useless functions: toLog2, inUnits && remove useless micro-definition 'DEV_IF_THROWS' && add ETH_TESTS flag to recover the DEV_INVARANIT_CHECK in both debug mode and test mode
3. add unit tests for Common.* of libdevcore

### CODE REVIEW

File: libdevcore/Common.h [hunk 1]
```diff
@@ -236,7 +224,7 @@ class InvariantChecker
 };
 
 /// Scope guard for invariant check in a class derived from HasInvariants.
-#if ETH_DEBUG
+#if (ETH_DEBUG || ETH_TESTS)
```
Comment (wheatli) [inline libdevcore/Common.h]: The DEBUG and TESTS macro are not relevant, we can not bind them to a logic here. If you need test exceptions throwing condition, you should add some unit test code under DEBUG macro.

File: test/unittests/libdevcore/Common.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    virtual bool invariants() const { return m_hasInvariants; }
+    void setHasInvariants(bool _hasInvariants) { m_hasInvariants = _hasInvariants; }
+    void check() { DEV_INVARIANT_CHECK_HERE; }
+
+private:
+    bool m_hasInvariants;
+};
+BOOST_AUTO_TEST_CASE(testCheckInvariants)
+{
+    FakeCheckInvariants fake_invariant;
```
Comment (wheatli) [inline test/unittests/libdevcore/Common.cpp]: Just like said upstair, plz make sure these checking logic are under DEBUG macro

Comment (cyjseagull) [reply test/unittests/libdevcore/Common.cpp]: already add `#if ETH_DEBUG, #endif` to this unit test.

File: libdevcore/Common.h (hunk 1 above)
Comment (cyjseagull) [reply libdevcore/Common.h]: already remove ETH_TESTS

**Pull request merged**